File: 1518.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        ans = numBottles
        while numBottles >= numExchange:
            ans += numBottles // numExchange
            numBottles = (numBottles // numExchange) + (numBottles % numExchange)
        return ans
    
if __name__ == '__main__':
    numBottles = int(input())
    numExchange = int(input())
    s = Solution()
    print(s.numWaterBottles(numBottles, numExchange))

File: 1518.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int ans = numBottles;
        while(numBottles / numExchange){
            ans += numBottles / numExchange;
            numBottles = (numBottles / numExchange) + (numBottles % numExchange);
        }
        return ans;
    }
};

int main() {
    Solution s;
    int numBottles, numExchange;
    cin >> numBottles >> numExchange;
    cout << s.numWaterBottles(numBottles, numExchange) << endl;
    return 0;
}